### PR TITLE
Revise PyTorch integration tutorial for improved readability and style

### PR DESCRIPTION
**Pull request by jburgh**
This commit introduces a significant revision to the PyTorch integration with W&B tutorial.

- Revised the title to be task-based and descriptive
- Added alt text for images
- Added a bit more context to "What this notebook covers"
- Added a "Background" section 
- Added context to the random block of code that previews the E2E process
- Reorganized content flow for a more logical progression
- Called out prerequisites and linked to PyTorch installation instructions
- Added troubleshooting link for authentication issues
- Clarified that the PyTorch-side setup was just a recommended approach
- Defeated the Kraken (it put up a fight!)
- Improved code samples with better comments and language tags
- Edited the example project name to be more intuitive (pytorch-demo)
- Highlighted that W&B conveniently provides dashboard URLs on init
- Changed "Define Training Logic" section to an H2 "Integrate W&B into your pipeline"  
- Shifted focus to the main W&B integration points watch/log
- Aligned terminology with PyTorch conventions (eg, "DataLoader")
- Deleted not-evergreen descriptions of the dashboard and linked to the Projects docs instead
- Created a What's next? section for advanced topics
- Checked and removed all the dead links a the end and added links to related content
- Edited for Google Dev Style 

Technical Qs flagged for dev review:

1. In the ConvNet class definition, there's a hardcoded value of `16` that might cause an error if the first kernel size is anything other than 16. Should it perhaps be:  nn.Conv2d(kernels[0], kernels[1], kernel_size=5, stride=1, padding=2),

2. In the test function, the ONNX export might not be ideal. Several review tools recommended to create a dedicated input tensor for ONNX export rather than using the last batch.

### CODE REVIEW

**Comment by jburgh** [comment]
Closing sample request
